md-setup/src/main/java/com/wipro/ats/bdre/md/setup/SetupDB.java

```diff
@@ -101,9 +99,9 @@ public void term() {
         session.getTransaction().rollback();
         session.close();
     }
-    private static Pattern pattern = Pattern.compile("\\$\\{(?<key>[^}]*)\\}");
 
-    public static String replaceVars(String line) throws Exception {
+
+    public static String replaceVars(String line) throws MetadataException {
         StringBuffer sb = new StringBuffer();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1149)